NeoMutt does not build with TDB in MacOS

<!--
Sorry you've found a bug, but thanks for reporting it here.
Please try to give as much information as you can.
-->

Hi, I'm the packager of NeoMutt for MacPorts.
I found an issue while trying to build it with TDB enabled.

This could be interpreted as an upstream bug in TDB and/or autosetup, but at the same time I think it would be nice to have it fixed in NeoMutt. 

## Expected Behaviour

`./configure --with-tdb=/opt/local` should detect TDB. 

<!-- What were you trying to do? -->

## Actual Behaviour

```
$ ./configure --with-tdb=/opt/local
...
Checking for tdb.h...not found
Error: Unable to find TDB
Try: 'configure --help' for options
```
From `config.log`:
```
Failed: /usr/bin/clang -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -arch x86_64 -std=c99 -D_ALL_SOU
RCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/opt/local/include -DNCURSES_WIDECHAR -c -std=c99 -pipe -Os -isysroot/Library/Developer/Co
mmandLineTools/SDKs/MacOSX10.14.sdk -arch x86_64 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/opt/local/include -g0
 conftest__.c -o conftest__.o -L/opt/local/lib
clang: warning: argument unused during compilation: '-L/opt/local/lib' [-Wunused-command-line-argument]
In file included from conftest__.c:2:
/opt/local/include/tdb.h:55:2: error: unknown type name 'size_t'; did you mean 'time_t'?
        size_t dsize;
        ^~~~~~
        time_t
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/sys/_types/_time_t.h:31:33: note: 'time_t' declared here
typedef __darwin_time_t         time_t;
                                ^
1 error generated.
child process exited abnormally
============
The failed code was:
#include <sys/stat.h>
#include <tdb.h>
int main(void) {

return 0;
}
============
```
Note that `size_it` is not found. Since `tdb.h` does not do so, programs including it should include `stddef.h` or `stdlib.h` beforehand. 

## Steps to Reproduce

See above.

## How often does this happen?

Always.

## When did it start to happen?

Release 20200417.

## NeoMutt Version

Could not build.

## Extra Info

<!-- This info might help us narrow down the source of the problem. -->

* Operating System and its version: MacOS 10.14.16, Xcode 11.3.1, MacPorts 2.6.2

--- a/auto.def
+++ b/auto.def
@@ -1058,7 +1058,7 @@ if {[get-define want-tdb]} {
     pkgconf true tdb
     define-feature tdb
   } else {
-    cc-with {-includes sys/stat.h} {
+    cc-with {-includes {sys/stat.h stddef.h} } {
       if {![check-inc-and-lib tdb [opt-val with-tdb $prefix] \
                               tdb.h tdb_close tdb]} {
         user-error "Unable to find TDB"

